joomla/joomla-cms
# [4.0] New UI Features for plg_system_httpheaders

### Summary of Changes

Based on on the awesome work done by @degobbis against my plugin this brings some more UI Configuration options to the plg_system_httpheaders plugin.

![image](https://user-images.githubusercontent.com/2596554/40052137-681234ae-583d-11e8-9d1d-31dc9fd98f73.png)

![image](https://user-images.githubusercontent.com/2596554/40053183-a6a39016-5840-11e8-9ef0-43a8bfd483bd.png)


### Testing Instructions

- apply this patch
- go to the plugin settings
- notice the new options
- enable the options for HSTS and make sure they work
- enable the csp header option (I would recommend report only!)
- check the browser console for the errors

### Expected result

HSTS & CSP now have some kind of UI inside of the plugin.

### Actual result

HSTS & CSP have no kind of UI inside of the plugin.

### Documentation Changes Required

The new options together with the plugin.

cc @SniperSister @mbabker @wilsonge @degobbis
